13-translations.Rmd

```diff
@@ -0,0 +1,71 @@
+particular focus on how things work for R, with the goal of making it as low-friction as possible for developers
+and users to contribute new/updated translations.
+
+## How translations work
+
+Each of the default packages distributed with R (i.e., those found in `./src/library` such as `base`, `utils`,
+and `stats`) contains a `po` directory that is the central location for cataloguing/translating each package's
+messages.
+
+### .pot files
+
+The `.pot` file is a snapshot of the messages available in a given **domain**. A domain in R typically identifies
+a source package and a source language (either R or C/C++). For example, the file `R-base.pot`
+(found in the R sources in `./src/library/base/po`) is a catalogue of all messages produced by R code in the
+`base` package, while `stats.pot` (_viz._, `./src/library/stats/po`) is a catalogue of all messages produced
+by C code in the `stats` package.
+
+There are two exceptions to the basic pattern described above. The first is the domain for C messages produced
+by the `base` package. Well, technically, they are not "produced by the `base` package" in the normal sense
+that code in the `src` directory of base is not where these messages come from, but rather, the C code which
+is the fundamental backing of R itself (especially, but not exclusively, the C code under `./src/main`).
+Given this idisyncrasy, the associated `.pot` file is `R.pot` and it's found in the `po` directory for `base`.
+
+The second is the domain for the Windows R GUI, i.e., the text in the menus and elsewhere in the R GUI program
+available for running R on Windows. These messages are stored in the `RGui.pot` domain, also in the `po`
+directory for `base`, and are most commonly derived from C code found in `./src/gnuwin32`. One reason to keep
+this domain separate is that it is only relevant to one platform (Windows). In particular, Windows has historically
+different character encodings, so that it made more sense for Windows developers to produce translations
+for Windows, since it is non-trivial for non-Windows users to test their translations for the Windows GUI.
+
+#### Generating .pot files
+
+For outside contributors, there's no need to update .pot files -- translators will typically take the R .pot files
+as given and generate .po files from them to be sent to an R-core maintainer as a patch.
+
+To emphasize, this section is almost always not needed for contributing translations -- it is here for
+completeness and edification.
+
+### .po files
+
+### .mo files
+
+.po files are plain text, but while helpful for human readers, this is inefficient for consumption by computers.
+The .mo format is a "compiled" version of the .po file optimized for retrieving messages when R is running.
+
+In R-devel, the conversion from .po to .mo is done by R-core -- you don't need to compile these files yourself.
+They are stored in the R sources at `./src/library/translations/inst` in various language-specific subdirectories.
+
```

Comment:
```suggestion

### Some useful tools

For being able to translate the messages in R, the following tools could be used. The [Poedit](https://poedit.net/) provides a GUI editor for writing translations in several language. An [online keyboard](https://www.lexilogos.com/keyboard/index.htm) could also help facilitate the translations process.

```

Reply:
I specifically avoided mentioning tools like Poedit because they are paid/proprietary software... does the guide have any official stance on implicit endorsement of non-open-source tooling?

Reply:
I don't think we want to endorse any non-open-source tooling. I did not realise it earlier! Let me exclude it from being mentioned in the guide.

Reply:
> there are also some tools available like `tools::checkPoFiles()`, `tools::update_pkg_po()`, ... Could these be useful for translators

My guess is "not much". Maybe we could use an advanced section where this comes up. But those functions are more useful for maintainers than for translators. Basically, those helpers are good for generating `.po` files, but `.po` files are the starting point for translators.

Reply:
> Does RStudio offer any specific features for working with .pot/.po?

Not that I know of.

Reply:
> Maybe write a sentence on naming conventions here or somewhere else?

I thought that was covered in the earlier paragraph:

> There are two exceptions to the basic pattern described above. The first is the domain for C messages produced
by the `base` package...

Let me know if we can improve the wording/flow.

Reply:
> Does the msginit look/behave the same way under Win?

Sorry I don't have access to any Windows machines to experiment...

Reply:
marking resolved since it's making a bit harder to follow the comment chain here, feel free to unresolve/add new comments on the latest version, thanks!